docs/src/pages/docs/network/02-state-management.mdx

```diff
@@ -0,0 +1,333 @@
+---
+title: "State Management in Agent Networks | Network Documentation | Mastra"
+description: Learn how to use state management in Mastra agent networks to share information between agents and maintain context throughout multi-step workflows.
+---
+
+# State Management in Agent Networks
+
+State management is a crucial feature of Agent Networks that allows agents to share information and maintain context throughout a multi-step workflow. The network state persists between agent calls, enabling complex interactions and data sharing without having to pass everything through inputs and outputs.
+
+## 1. Network State Basics
+
+The `NetworkState` class provides a simple key-value store that persists throughout the network execution:
+
+```ts showLineNumbers copy
+import { AgentNetwork, NetworkState } from "@mastra/core/network";
+
+// Create a network with initial state
+const network = new AgentNetwork({
+  name: "Stateful Network",
+  agents: [agentA, agentB],
+  routingModel: openai("gpt-4o"),
+  initialState: new NetworkState({
+    counter: 0,
+    items: [],
+    user: { name: "Alice", preferences: { theme: "dark" } },
+  }),
```

Comment:
If we wrapped agents as tool calls in the router agent, wouldn't we not need this? Since the router agent would maintain state via input/output tool schemas in context tokens